pkg/internal/codegen/parse/crd.go

```diff
@@ -65,14 +65,24 @@ func (b *APIs) parseCRDs() {
 							Group:   fmt.Sprintf("%s.%s", resource.Group, resource.Domain),
 							Version: resource.Version,
 							Names: v1beta1.CustomResourceDefinitionNames{
-								Kind:   resource.Kind,
-								Plural: resource.Resource,
+								Kind:       resource.Kind,
+								Plural:     resource.Resource,
+								ShortNames: []string{resource.ShortName},
 							},
 							Validation: &v1beta1.CustomResourceValidation{
 								OpenAPIV3Schema: &resource.JSONSchemaProps,
 							},
 						},
 					}
+
+					result, err := parseResourceAnnotation(resource.Type)
+					if err != nil {
```

Comment:
this code assumes the annotation will always exists ? which isn't the case for existing projects.

Reply:
If you look at this linehttps://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/crd.go#L41
which concludes the following thing: https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/util.go#L59
If I'm not wrong, it assumes that every CRD will have a `resource` annotation. That's the only reason why I didn't include any condition for this code. Let me know your inputs for this.

Reply:
It assumes *either* you have a resource annotation *or* you have TypeMeta and ObjectMeta.  The comment on the function isn't clear -- it should probably be clarified (you can do so in a separate commit in this PR).